tools/src/main/java/org/apache/kafka/trogdor/coordinator/TaskManager.java

```diff
@@ -142,13 +145,15 @@
             Utils.join(nodeManagers.keySet(), ", "));
     }
 
-    enum ManagedTaskState {
+    public enum ManagedTaskState {
```

Comment:
Hmm.  This is internal, and we shouldn't expose it.  Instead, we should use the stuff that's public, in `tools/src/main/java/org/apache/kafka/trogdor/rest/`.

We already have `TaskState`:
```
/**
 * The state which a task is in on the Coordinator.
 */
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME,
    include = JsonTypeInfo.As.PROPERTY,
    property = "state")
@JsonSubTypes({
        @JsonSubTypes.Type(value = TaskPending.class, name = "PENDING"),
        @JsonSubTypes.Type(value = TaskRunning.class, name = "RUNNING"),
        @JsonSubTypes.Type(value = TaskStopping.class, name = "STOPPING"),
        @JsonSubTypes.Type(value = TaskDone.class, name = "DONE")
    })
public abstract class TaskState extends Message {
```

How about creating a `TaskStateType` enum which could be either PENDING, RUNNING, STOPPING, DONE?  Then the `TaskState` annotation can reference those enum constants.  You can also use something like `Option<TaskStateType>` to nicely express all this (no more parsing strings yourself).